### DISCUSSION TITLE
custom OIDC as

### DISCUSSION THREAD

**Discussion by altounsi** [keycloak/keycloak#11115] [Q&A]
I want to use Keycloak as an identity broker and I want to send these custom parameters to the IdP. could you guide me on where to do the change (what classes ) with steps if possible I would really appreciate it
scope: openid
response_type: id_token
response_mode: form_post
client_id: https://me-as-broker
redirect_uri: my oidc reponse URL 
nonce: uuid
ui_locales: 
aud: https://client.com
max_age: number
state: uuid

**Comment by altounsi** [comment]
anyone

**Reply by fidgi** [reply]
Hi,
using Keycloak as an identity broker is when you configure keycloak to delegate the identification to a another IdP (as explained here https://www.keycloak.org/docs/latest/server_admin/index.html#_identity_broker).

It seems it's the reverse that you want. 
If this is the case then you just have to register an OIDC client for the IdP to be able to forward authentification request to keycloak . 
See https://www.keycloak.org/docs/latest/server_admin/#_oidc_clients

hope this will help

**Reply by altounsi** [reply]
ُThank you so much for trying to help me here,

What I want is the identity broker option, however, the Idp that I delegate the identification to is not following the OIDC standard as is, 
he is using some of the standard parameters and some custom ones of his own so I want to forward these custom parameters to him. 

Thanks again

**Comment by altounsi** [comment]
anyone

**Comment by altounsi** [comment]
anyone

**Comment by altounsi** [comment]
anyone

**Comment by altounsi** [comment]
UP

**Reply by Underknowledge** [reply]
FYI, not moving up, found you on page 3

**Comment by altounsi** [comment]
more up
